File: frontend/src/types/Preview.ts
export interface Preview {
  id?: string;
  job_id: string;
  slot_id: string;
  status: string;
  output_video_path?: string;
  download_path?: string;
  duration_seconds?: number;
  render_retry_count?: number;
  error_code?: string | null;
  error_message?: string | null;
  created_at?: string;
  artifact_manifest?: Record<string, unknown>;
  render_metrics?: Record<string, unknown>;
  completed_at?: string | null;
}

File: backend/internal/models/preview.go
package models

type Preview struct {
	ID               string   `json:"id,omitempty"`
	JobID            string   `json:"job_id"`
	SlotID           string   `json:"slot_id"`
	Status           string   `json:"status"`
	OutputVideoPath  string   `json:"output_video_path,omitempty"`
	DownloadPath     string   `json:"download_path,omitempty"`
	DurationSeconds  float64  `json:"duration_seconds,omitempty"`
	CreatedAt        string   `json:"created_at,omitempty"`
	CompletedAt      *string  `json:"completed_at"`
	ArtifactManifest Metadata `json:"artifact_manifest,omitempty"`
	RenderMetrics    Metadata `json:"render_metrics,omitempty"`
}
